### PR TITLE
[Owner Review] Added NiDmm missing enum metadata and auto generated them

### PR DESCRIPTION
**Pull request by ShantanuShinde**
### What does this Pull Request accomplish?

The metadata of missing enums and autogenerted them
The name and values of the enums were obtained from the nidmm.h file and they were added to the dictionary in enums.py following the format of other enums. The enums were then referenced in the attributes and functions where they are being used in the attributes.py and functions.py respectively

### Why should this Pull Request be merged?

The missing enums can now be autogenerated

### Testing

The build was successful and the enums got added to the nidmm.proto file

### CODE REVIEW

**Review by ganeshnie — APPROVED** [review]

File: source/codegen/metadata/nidmm/attributes.py [hunk 1]
```diff
@@ -768,6 +771,7 @@
         },
         'enum': 'AutoZero',
         'lv_property': 'Configuration:Measurement Options:Auto Zero',
+        'enum': 'AutoZero',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/attributes.py]: `'enum': 'AutoZero'` is already specified two lines up.

File: source/codegen/metadata/nidmm/enums.py [hunk 2]
```diff
@@ -68,48 +44,30 @@
...
-                },
                 'name': 'NIDMM_VAL_SECONDS',
                 'value': 0
             },
             {
-                'documentation': {
-                    'description': 'Powerline Cycles'
-                },
-                'name': 'NIDMM_VAL_POWER_LINE_CYCLES',
+               'name': 'NIDMM_VAL_POWER_LINE_CYCLES',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 3]
```diff
@@ -68,48 +44,30 @@
...
-                },
                 'name': 'NIDMM_VAL_AUTO_ZERO_OFF',
                 'value': 0
             },
             {
-                'documentation': {
-                    'description': 'The DMM internally disconnects the input signal following each measurement  and takes a zero reading. It then subtracts the zero reading from the  preceding reading.'
-                },
-                'name': 'NIDMM_VAL_AUTO_ZERO_ON',
+               'name': 'NIDMM_VAL_AUTO_ZERO_ON',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 4]
```diff
@@ -118,62 +76,86 @@
...
-                },
                 'name': 'NIDMM_VAL_CABLE_COMP_SHORT',
                 'value': 2
             },
             {
-                'documentation': {
-                    'description': 'Open and Short Cable Compensation'
-                },
-                'name': 'NIDMM_VAL_CABLE_COMP_OPEN_AND_SHORT',
+               'name': 'NIDMM_VAL_CABLE_COMP_OPEN_AND_SHORT',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 5]
```diff
@@ -182,107 +164,62 @@
...
-                },
                 'name': 'NIDMM_VAL_DC_CURRENT',
                 'value': 3
             },
             {
-                'documentation': {
-                    'description': 'AC Current'
-                },
-                'name': 'NIDMM_VAL_AC_CURRENT',
+               'name': 'NIDMM_VAL_AC_CURRENT',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 6]
```diff
@@ -182,107 +164,62 @@
...
                 'value': 4
             },
             {
-                'documentation': {
-                    'description': '2-Wire Resistance'
-                },
-                'name': 'NIDMM_VAL_2_WIRE_RES',
+               'name': 'NIDMM_VAL_2_WIRE_RES',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 7]
```diff
@@ -182,107 +164,62 @@
...
-                },
                 'name': 'NIDMM_VAL_4_WIRE_RES',
                 'value': 101
             },
             {
-                'documentation': {
-                    'description': 'Frequency'
-                },
-                'name': 'NIDMM_VAL_FREQ',
+               'name': 'NIDMM_VAL_FREQ',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 8]
```diff
@@ -291,104 +228,94 @@
...
-                },
                 'name': 'NIDMM_VAL_NONE',
                 'value': -1
             },
             {
-                'documentation': {
-                    'description': 'AUX I/O Connector'
-                },
-                'name': 'NIDMM_VAL_EXTERNAL',
+               'name': 'NIDMM_VAL_EXTERNAL',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 9]
```diff
@@ -291,104 +228,94 @@
...
-                },
                 'name': 'NIDMM_VAL_PXI_TRIG4',
                 'value': 115
             },
             {
-                'documentation': {
-                    'description': 'PXI Trigger Line 5'
-                },
-                'name': 'NIDMM_VAL_PXI_TRIG5',
+               'name': 'NIDMM_VAL_PXI_TRIG5',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 10]
```diff
@@ -397,87 +324,122 @@
     'MeasurementDestinationSlope': {
         'values': [
             {
-                'documentation': {
-                    'description': 'Rising Edgs'
-                },
-                'name': 'NIDMM_VAL_POSITIVE',
+               'name': 'NIDMM_VAL_POSITIVE',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 11]
```diff
@@ -397,87 +324,122 @@
...
+        ]
+    },
     'OperationMode': {
         'values': [
             {
-                'documentation': {
-                    'description': 'IviDmm Mode'
-                },
-                'name': 'NIDMM_VAL_IVIDMM_MODE',
+               'name': 'NIDMM_VAL_IVIDMM_MODE',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

File: source/codegen/metadata/nidmm/enums.py [hunk 12]
```diff
@@ -397,87 +324,122 @@
...
                 'value': 0
             },
             {
-                'documentation': {
-                    'description': 'Waveform acquisition mode'
-                },
-                'name': 'NIDMM_VAL_WAVEFORM_MODE',
+               'name': 'NIDMM_VAL_WAVEFORM_MODE',
```
Comment (reckenro) [inline source/codegen/metadata/nidmm/enums.py]: The spacing changed for 'name'. Make it line up with 'value' beneath.

**Review by reckenro — APPROVED** [review]
Looks good overall. Just some formatting to clean up.

File: generated/nidmm/nidmm.proto [hunk 13]
```diff
@@ -229,6 +229,34 @@ enum CableCompensationType {
...
+enum CompensatedOhms {
+  option allow_alias = true;
+  COMPENSATED_OHMS_UNSPECIFIED = 0;
+  COMPENSATED_OHMS_NIDMM_VAL_OFFSET_COMP_OHMS_OFF = 0;
+  COMPENSATED_OHMS_NIDMM_VAL_OFFSET_COMP_OHMS_ON = 1;
+}
+
+enum DCBias {
+  option allow_alias = true;
+  D_C_BIAS_UNSPECIFIED = 0;
```
Comment (jonlugoatx) [inline generated/nidmm/nidmm.proto]: The `D_C_` should be `DC`. Same with `L_C_` below.

**Review by jonlugoatx — APPROVED** [review]
1. Please do not remove the testing section from your PR description. That template is there for a reason. If you have done no testing, then say so in that section. In this case, I would expect you to at least say that it builds and runs successfully.
2. You did not comment anywhere on your process for getting these enum values. Did you and copy them from the base enums.hapigen and do hand conversions?
3. How did you arrive at the changes to functions.py? Was that information in the hapigen already but not the nimi-python?

I marked as approved to save time since the changes look good (with a little bit of feedback), but please post responses to these questions. If I have additional feedback based on your answers, then we can address them in a follow-up PR.

**Comment by ShantanuShinde** [comment]
> 
> 
>     1. Please do not remove the testing section from your PR description. That template is there for a reason. If you have done no testing, then say so in that section. In this case, I would expect you to at least say that it builds and runs successfully.
> 
>     2. You did not comment anywhere on your process for getting these enum values. Did you and copy them from the base enums.hapigen and do hand conversions?
> 
>     3. How did you arrive at the changes to functions.py? Was that information in the hapigen already but not the nimi-python?
> 
> 
> I marked as approved to save time since the changes look good (with a little bit of feedback), but please post responses to these questions. If I have additional feedback based on your answers, then we can address them in a follow-up PR.

@jonlugoatx 
I added back the testing section in this as well as the other attributes pr and also added the description of how the enums and attributes were added

File: generated/nidmm/nidmm.proto (hunk 13 above)
Comment (ShantanuShinde) [reply generated/nidmm/nidmm.proto]: this is autogenerated to D_C_. same is the case for the attributes that were already present with DC or LC names. should I change this by editing nidmm.proto manually?

File: source/codegen/metadata/nidmm/attributes.py (hunk 1 above)
Comment (ShantanuShinde) [reply source/codegen/metadata/nidmm/attributes.py]: fixed

File: source/codegen/metadata/nidmm/enums.py (hunk 2 above)
Comment (ShantanuShinde) [reply source/codegen/metadata/nidmm/enums.py]: fixed

File: source/codegen/metadata/nidmm/enums.py (hunk 3 above)
Comment (ShantanuShinde) [reply source/codegen/metadata/nidmm/enums.py]: fixed

File: generated/nidmm/nidmm.proto (hunk 13 above)
Comment (ShantanuShinde) [reply generated/nidmm/nidmm.proto]: replaced DC with Dc so now DC is generated. done similar for LC, ADCC, and RTD which have similar generation

**Review by jonlugoatx — APPROVED** [review]
Be sure to run the system tests before submitting.

**Pull request merged**